### PR TITLE
[1LP][RFR] Collection of SystemImageType

### PR DESCRIPTION
**Pull request by anurag03**
Purpose or Intent
Converting the SystemImageType to collection

{{pytest: cfme/tests/infrastructure/test_system_image_type.py -v }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit cb0ca7365479a94952a793a506f466d7fc7aea43:

`cfme/tests/infrastructure/test_system_image_type.py`:
- :heavy_exclamation_mark: Line 68:1: **W391** *blank line at end of file*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/infrastructure/test_system_image_type.py [hunk 1]
```diff
@@ -9,41 +9,49 @@
...
+def collection_init(appliance):
+    return appliance.collections.system_image_type
+
+
+def test_system_image_type_crud(collection_init):
     """
     Tests a System Image Type using CRUD operations.
     """
-    sys_image_type = SystemImageType(
+    collection = collection_init
```
Comment (quarckster) [inline cfme/tests/infrastructure/test_system_image_type.py]: **REQUIRED**
Please either rename `collection_init` fixture or delete this redundant line.

File: cfme/infrastructure/pxe.py [hunk 2]
```diff
@@ -591,6 +594,7 @@ class PXESystemImageTypesView(PXEMainView):
     """
     represents whole All System Image Types page
     """
+    table = Table('records_div')
```
Comment (quarckster) [inline cfme/infrastructure/pxe.py]: What's the purpose of this change in this PR?

Comment (anurag03) [reply cfme/infrastructure/pxe.py]: I included that in this PR mistakenly, thanks for highlighting it.

File: cfme/infrastructure/pxe.py [hunk 3]
```diff
@@ -644,29 +648,12 @@ class SystemImageType(Updateable, Pretty, Navigatable):
...
-        view.fill({'name': self.name, 'type': self.provision_type})
-        main_view = self.create_view(PXESystemImageTypesView)
-        if cancel:
-            view.cancel.click()
-            msg = 'Add of new System Image Type was cancelled by the user'
-        else:
-            view.add.click()
-            msg = 'System Image Type "{}" was added'.format(self.name)
-        main_view.flash.assert_success_message(msg)
+    # def __init__(self, name=None, provision_type=None, appliance=None):
```
Comment (psav) [inline cfme/infrastructure/pxe.py]: **REQUIRED:** Please remove these comment lines

File: cfme/infrastructure/pxe.py [hunk 4]
```diff
@@ -704,6 +691,39 @@ def delete(self, cancel=True):
...
+            provision_type: Type on Image. i.e Vm and Instance or Host
+            cancel (boolean): Whether to cancel out of the creation.  The cancel is done
+                after all the information present in the SIT has been filled in the UI.
+        """
+
+        system_image_type = self.instantiate(name, provision_type)
+
+        view = navigate_to(self, 'Add')
+        view.fill({'name': name, 'type': provision_type})
+        main_view = self.create_view(PXESystemImageTypesView)
```
Comment (psav) [inline cfme/infrastructure/pxe.py]: **REQUIRED:** To assist in the flow of reading, ```main_view``` definition should be moved below the conditional

File: cfme/tests/infrastructure/test_system_image_type.py [hunk 5]
```diff
@@ -9,41 +9,46 @@
 pytestmark = [pytest.mark.tier(3)]
 
 
-def test_system_image_type_crud():
+@pytest.fixture(scope="module")
+def collection(appliance):
```
Comment (psav) [inline cfme/tests/infrastructure/test_system_image_type.py]: **REQUIRED:** This is too generic a name, I actually don't think a fixture is necessary here and would prefer to use the appliance call each time in the test, requiring the appliance fixture instead.

File: cfme/infrastructure/pxe.py (hunk 3 above)
Comment (anurag03) [reply cfme/infrastructure/pxe.py]: I missed that, thanks for highlighting 👍

File: cfme/tests/infrastructure/test_system_image_type.py (hunk 5 above)
Comment (quarckster) [reply cfme/tests/infrastructure/test_system_image_type.py]: @psav I would be thankful if you explain the point of using appliance call each time.

File: cfme/infrastructure/pxe.py [hunk 6]
```diff
@@ -704,6 +687,39 @@ def delete(self, cancel=True):
...
+        else:
+            view.add.click()
+            msg = 'System Image Type "{}" was added'.format(name)
+        main_view = self.create_view(PXESystemImageTypesView)
+        main_view.flash.assert_success_message(msg)
+        return system_image_type
+
+
+@navigator.register(SystemImageTypeCollection, 'All')
 @navigator.register(SystemImageType, 'All')
```
Comment (mshriver) [inline cfme/infrastructure/pxe.py]: **Required**

Remove the registration against SystemImageType, the entity object doesn't have an 'All' page

File: setup.py [hunk 7]
```diff
@@ -97,6 +97,7 @@
             'clusters = cfme.infrastructure.cluster:ClusterCollection',
             'hosts = cfme.infrastructure.host:HostCollection',
             'deployment_roles = cfme.infrastructure.deployment_roles:DeploymentRoleCollection',
+            'system_image_type = cfme.infrastructure.pxe:SystemImageTypeCollection',
```
Comment (mshriver) [inline setup.py]: **Required**

The key has to be plural, matching all of the others.

File: cfme/infrastructure/pxe.py (hunk 6 above)
Comment (anurag03) [reply cfme/infrastructure/pxe.py]: @mshriver: It is required for details page navigation.

Comment (mshriver) [reply cfme/infrastructure/pxe.py]: How is it required?

Navigating to details can be done with a prerequisite of `NavigateToAttribute('parent', 'All'), since the entity object has a 'parent' attribute that is the collection object.
